src/main/java/com/redhat/labs/lodestar/service/GitService.java

```diff
@@ -0,0 +1,337 @@
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitService.class);
+
+	private static final String ARTIFACT_FILE = "artifacts.json";
+	private static final String ENGAGEMENT_FILE = "engagement.json";
+
+	@ConfigProperty(name = "group.parent.id")
+	Integer groupParentId;
+
+	@ConfigProperty(name = "default.branch")
+	String defaultBranch;
+
+	@ConfigProperty(name = "default.commit.message")
+	String defaultCommitMessage;
+
+	@ConfigProperty(name = "default.author.name")
+	String defaultAuthorName;
+
+	@ConfigProperty(name = "default.author.email")
+	String defaultAuthorEmail;
+
+	@ConfigProperty(name = "default.page.size")
+	Integer pageSize;
+
+	@Inject
+	@RestClient
+	GitlabRestClient gitlabRestClient;
+
+	@Inject
+	Jsonb jsonb;
+
+	/**
+	 * Returns the name of the default branch for the given project ID. If not
+	 * found, the configured value is returned.
+	 * 
+	 * @param projectId
+	 * @return
+	 */
+	String findDefaultBranch(Integer projectId) {
+
+		List<Branch> branches = gitlabRestClient.findProjectBranches(projectId);
+		Optional<Branch> branch = branches.stream().filter(Branch::getIsDefault).findAny();
+
+		return branch.isPresent() ? branch.get().getName() : defaultBranch;
+
+	}
+
+	/**
+	 * Creates a {@link ProjectTree} for the given {@link Project}.
+	 * 
+	 * @param project
+	 * @return
+	 */
+	ProjectTree createProjectTree(Project project) {
+
+		LOGGER.debug("creating project tree for {}", project);
+
+		List<ProjectTreeNode> treeNodes = getProjectTree(project.getId(), false);
+		return ProjectTree.builder().projectId(project.getId()).projectTreeNodes(treeNodes).build();
+
+	}
+
+	/**
+	 * Returns an {@link Optional} containing the {@link ProjectTreeNode} for the
+	 * artifacts.json file. Otherwise, an empty {@link Optional} is returned.
+	 * 
+	 * @param projectTree
+	 * @return
+	 */
+	Optional<File> getArtifactsFile(ProjectTree projectTree) {
+
+		Optional<ProjectTreeNode> artifactsTreeNode = projectTree.getArtifactsProjectTreeNode();
+		if (artifactsTreeNode.isEmpty()) {
+			return Optional.empty();
+		}
+
+		// get default branch
+		String branch = findDefaultBranch(projectTree.getProjectId());
+
+		return Optional.ofNullable(
+				gitlabRestClient.getFile(projectTree.getProjectId(), artifactsTreeNode.get().getPath(), branch));
+
+	}
+
+	/*
+	 * Returns a {@link List} of {@link Artifact}s for the given {@link File}. An
+	 * empty {@link List} is returned if the {@link File} is not found or contains
+	 * no {@link Artifact}s.
+	 */
+	List<Artifact> parseFile(Optional<File> file) {
+
+		List<Artifact> artifacts = new ArrayList<>();
+
+		if (file.isEmpty()) {
+			return artifacts;
+		}
+
+		File artifactsFile = file.get();
+
+		artifactsFile.decodeFileAttributes();
+
+		if (ARTIFACT_FILE.equals(artifactsFile.getFilePath())) {
+			artifacts = parseArtifactsFile(artifactsFile);
+		} else if (ENGAGEMENT_FILE.equals(artifactsFile.getFilePath())) {
+			artifacts = parseEngagementFile(artifactsFile);
```

Comment:
was hoping to leave all engagement.json queries to git-api

Reply:
no longer handling engagement.json